bin/deploy-vhost

```diff
@@ -1056,6 +1057,45 @@ sub remove_site {
 
 }
 
+# Upgrade database password on the fly
+sub upgrade_db_password {
+    my ($params, $database) = @_;
+
+    my $db_host = get_db_host ($params, 0);
+
+    if ($params->{type} eq 'psql') {
+        # Try to connect with the old password.  If we can't, assume upgrade already done.
+        my $dbh = DBI->connect("dbi:Pg:dbname=$database;host=$db_host;port=$params->{port}", $database, pgpw($database), { PrintError => 0, RaiseError => 0 });
+        if ($dbh) {
+            # Set password to the new type
+            if (!($dbh->do("ALTER ROLE \"$database\" PASSWORD \"".pgpw("-n $database")."\";"))) {
+                warn "unable to upgrade password for postgres database $database";
+            }
+            $dbh->disconnect();
+        } else {
+            if (DBI->errstr() !~ /password authentication failed/) {
+                warn "unexpected error trying to connect to $database: ".DBI->errstr();
+            }
+        }
+    } elsif ($params->{type} eq 'mysql') {
+        # Try to connect with the old password.  If we can't, assume upgrade already done.
+        my $dbh = DBI->connect("dbi:mysql:dbname=$database;host=$db_host", $database, pgpw($database), { PrintError => 0, RaiseError => 0 });
+        if ($dbh) {
+            # Set password to the new type
+            print "SET PASSWORD = PASSWORD('".pgpw($database, 1)."');\n";
```

Comment:
Not sure what this print line is for. (Plus it's got a second argument to pgpw?)

Reply:
Oops, missed that. Thanks :)